src/components/forms/AutoComplete/index.tsx

```diff
@@ -1,74 +1,146 @@
-import React, {useEffect, useState} from "react";
-import { TextField } from "@material-ui/core";
+import React, { useEffect, useState } from 'react';
+import { TextField } from '@material-ui/core';
 import Autocomplete from '@material-ui/lab/Autocomplete';
-import Utils from "../../../helpers/utils";
-import TextInput from "../TextInput";
+import Utils from '../../../helpers/utils';
+import TextInput from '../TextInput';
 import isDeepEqual from 'fast-deep-equal/react';
 
+declare const PCore;
 
 interface IOption {
   key: string;
   value: string;
 }
 
-export default function AutoComplete(props) {
+export const getDisplayFieldsMetaData = columns => {
+  const displayColumns = columns.filter(col => col.display === 'true');
+  const metaDataObj: any = { key: '', primary: '', secondary: [] };
+  const keyCol = columns.filter(col => col.key === 'true');
+  metaDataObj.key = keyCol.length > 0 ? keyCol[0].value : 'auto';
+  for (let index = 0; index < displayColumns.length; index += 1) {
+    if (displayColumns[index].primary === 'true') {
+      metaDataObj.primary = displayColumns[index].value;
+    } else {
+      metaDataObj.secondary.push(displayColumns[index].value);
+    }
+  }
+  return metaDataObj;
+};
+
+export const preProcessColumns = columns => {
```

Comment:
I'm not sure why this function is outside the Autocomplete function scope. It doesn't seem to be used by other components. Should it be inside Autocomplete?